File: internal/dto/clusterrole.go
package dto

type PolicyRule struct {
	Resources       []string
	Verbs           []string
	APIGroups       []string
	ResourceNames   []string
	NonResourceURLs []string
}

type ClusterRoleRef struct {
	Name string `json:"name"`
}

type ClusterRole struct {
	Name          string
	Age           string
	CreatedAt     string
	Labels        map[string]string
	Annotations   map[string]string
	ManagedFields []ManagedField
	Rules         []PolicyRule
}

File: packages/core/frontend/src/types/resources/clusterrole.ts
import type { ManagedField } from "./shared";

export interface PolicyRule {
  Resources: string[];
  Verbs: string[];
  APIGroups: string[];
  ResourceNames: string[];
  NonResourceURLs: string[];
}

export interface ClusterRole {
  Name: string;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  Rules: PolicyRule[];
}
